### PR TITLE
Refactor chat state

### PR DESCRIPTION
**Pull request by gtalha07**
- [x] Redirect ```flutter_chat_ui``` to upstream, update core acter packages for Flutter 3.10 compatibility.
- [x] Rewrite chat list state listeners.
- [x] Rewrite chat room state listeners.
- [x] Align chat states with the UI.
- [x] Update chat theme to be friendly and intuitive.
- [x] Activate mentions and improvements.

**Pending fixes**
- Image message send is not working.
- Multiple images selection and send is not working.
- ```Replying To``` typos.
- Chat room profile section.
- Chat bubble overflows in some scenarios.
- Bubble swipe to action for message actions.

Previews:
Screenshot 2023-07-27 at 12 51 07 AM
Screenshot 2023-07-27 

at 12 51 34 AM

### CODE REVIEW

**Comment by bitfriend** [comment]
When opened chat room, I met too many of the following logs and couldn't click by mouse anywhere.
```
[ERROR:flutter/shell/gpu/gpu_surface_gl_skia.cc(217)] Could not make the context current to acquire the frame.

════════ Exception caught by scheduler library ═════════════════════════════════
'package:flutter/src/rendering/mouse_tracker.dart': Failed assertion: line 195 pos 12: '!_debugDuringDeviceUpdate': is not true.
════════════════════════════════════════════════════════════════════════════════
[ERROR:flutter/shell/gpu/gpu_surface_gl_skia.cc(217)] Could not make the context current to acquire the frame.

════════ Exception caught by scheduler library ═════════════════════════════════
'package:flutter/src/rendering/mouse_tracker.dart': Failed assertion: line 195 pos 12: '!_debugDuringDeviceUpdate': is not true.
════════════════════════════════════════════════════════════════════════════════

════════ Exception caught by scheduler library ═════════════════════════════════
'package:flutter/src/rendering/mouse_tracker.dart': Failed assertion: line 195 pos 12: '!_debugDuringDeviceUpdate': is not true.
════════════════════════════════════════════════════════════════════════════════
```
This issue disappeared after [this commit](https://github.com/acterglobal/a3/pull/765/commits/261f906185e0c1f4ef20d7271a82d35cad1ad813).

**Review by bitfriend — APPROVED** [review]
Looks good to me

File: app/lib/features/chat/widgets/convo_list.dart [hunk 1]
```diff
@@ -1,100 +1,99 @@
...
         heightFactor: 10,
-        child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            Text('${AppLocalizations.of(context)!.loadingConvo}...'),
-            const SizedBox(height: 10),
-            const CircularProgressIndicator(),
-          ],
+        child: Text(
+          '${AppLocalizations.of(context)!.loadingConvo}...',
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/convo_list.dart:22]: shouldn't that `...` also be in there, if we already use it...?

```suggestion
          AppLocalizations.of(context)!.loadingConvo,
```

**Review by gnunicorn — APPROVED** [review]
Great work, mate! Looks good overall, just a few nitpicks. so excited to finally try this!

**Review by gnunicorn — APPROVED** [review]

File: app/lib/features/chat/providers/notifiers/chat_room_notifier.dart [hunk 2]
```diff
@@ -0,0 +1,981 @@
...
+  late String roomId;
+
+  ChatRoomNotifier({
+    required this.ref,
+  }) : super(const ChatRoomState.loading());
+
+  void init(String id) async {
+    client = ref.watch(clientProvider)!;
+    roomId = id;
+    if (roomId.isNotEmpty) {
```
[the remark this answers is not in the document]
Comment (gtalha07) [reply app/lib/features/chat/providers/notifiers/chat_room_notifier.dart:52]: if you happen to go to ```RoomPage``` you'll find this state call
```
 @override
  void initState() {
    super.initState();
    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
      ref.watch(chatRoomProvider.notifier).init(widget.convo.getRoomIdStr());
      ref.watch(chatRoomProvider.notifier).fetchUserProfiles();
    });
  }
```
the ```roomId``` provider reactively updates the ```ChatRoomNotifier```. I agree the approach isn't very safe but I'll add try catch call to update the state correctly.

File: app/lib/features/chat/providers/notifiers/chat_room_notifier.dart [hunk 3]
```diff
@@ -0,0 +1,981 @@
...
+      room = await ref.read(chatProvider(roomId).future);
+      timeline = await room.timelineStream();
+      StreamSubscription<TimelineDiff>? subscription;
+      subscription = timeline.diffRx().listen((event) {
+        _parseEvent(event);
+      });
+      while (ref.read(messagesProvider).length < 10) {
+        await timeline.paginateBackwards(10);
+      }
```
[the remark this answers is not in the document]
Comment (gtalha07) [reply app/lib/features/chat/providers/notifiers/chat_room_notifier.dart:61]: We need to fix returning boolean from this call as it always stays true....

File: app/lib/features/chat/providers/notifiers/chat_room_notifier.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (gtalha07) [reply app/lib/features/chat/providers/notifiers/chat_room_notifier.dart:52]: Ahh I missed it. It'll never reach to empty state. We are firing this method with included param ```roomId```
```
WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
      ref.watch(chatRoomProvider.notifier).init(widget.convo.getRoomIdStr());
      ref.watch(chatRoomProvider.notifier).fetchUserProfiles();
    });
    ``` 
 Thanks for pointing it out. The ```if``` ```else``` case is outdated for this iteration. I'll remove it

File: app/lib/features/chat/widgets/convo_list.dart (hunk 1 above)
Comment (gtalha07) [reply app/lib/features/chat/widgets/convo_list.dart:22]: Its not in there.

**Pull request merged**